docs/kubernetes/deploy.md

```diff
@@ -0,0 +1,130 @@
+Heapster is running at https://contoso-apple-59769a59.westus2.cloudapp.azure.com/api/v1/proxy/namespaces/kube-system/services/heapster
+KubeDNS is running at https://contoso-apple-59769a59.westus2.cloudapp.azure.com/api/v1/proxy/namespaces/kube-system/services/kube-dns
+kubernetes-dashboard is running at https://contoso-apple-59769a59.westus2.cloudapp.azure.com/api/v1/proxy/namespaces/kube-system/services/kubernetes-dashboard
+
+To further debug and diagnose cluster problems, use 'kubectl cluster-info dump'.
+```
+
+<a href="#the-long-way"></a>
+
+## ACS Engine the Long Way
+
+### Step 1: Generate an SSH Key
+
+In addition to using Kubernetes APIs to interact with the clusters, cluster operators may access the master and agent machines using SSH.
+
+If you don't have an SSH key [cluster operators may generate a new one](../ssh.md#ssh-key-generation).
+
+### Step 2: Create a Service Principal
+
+Kubernetes clusters have integrated support for various cloud providers as core functionality. On Azure, acs-engine uses a Service Principal to interact with Azure Resource Manager (ARM). Follow the instructions to [create a new service principal](../serviceprincipal.md)
+
+### Step 3: Edit your Cluster Definition
+
+ACS Engine consumes a cluster definition which outlines the desired shape, size, and configuration of Kubernetes. There are a number of features that can be enabled through the cluster definition, check the `examples` directory for a number of... examples.
+
+Edit the [simple Kubernetes cluster definition](../examples/kubernetes.json) and fill out the required values:
+
+* `dnsPrefix`: must be a region-unique name and will form part of the hostname (e.g. myprod1, staging, leapinglama), be unique!
+* `keyData`: must contain the public portion of an SSH key, this will be associated with the `adminUsername` value found in the same section of the cluster definition (e.g. 'ssh-rsa AAAAB3NzaC1yc2EAAAADAQABA....')
+* `servicePrincipalClientID`: this is the appId uuid or name from step 3
+* `servicePrincipalClientSecret`: this is the password or randomly-generated password from step 3
+
+### Step 4: Generate the Templates
+
+The generate command takes a cluster definition and outputs a number of templates which describe your Kubernetes cluster. By default, `generate` will create a new directory named after your cluster nested in the `_output` directory. If my dnsPrefix was `larry` my cluster templates would be found in `_output/larry-`.
+
+Run `acs-engine generate examples/kubernetes.json`
+
+### Step 5: Submit your Templates to Azure Resource Manager (ARM)
+
+6. [deploy the output azuredeploy.json and azuredeploy.parameters.json](../acsengine.md#deployment-usage)
+  * To enable the optional network policy enforcement using calico, you have to
+    set the parameter during this step according to this [guide](../kuberntes.md#optional-enable-network-policy-enforcement-using-calico)
+
+
+### Step 6: If using an existing VNET
```

Comment:
This step is required the long way or the short way.  How do we surface it in both places?  This is a common support question.